### DISCUSSION TITLE
GUI doesn't use MQTT commands for controlling blinds?

### DISCUSSION THREAD

**Discussion by jenniferlee1818** [arendst/Tasmota#15250] [Q&A] [answered]
It seems like the MQTT broker would receive _ShutterOpen_, _ShutterClose_ or _ShutterStop_ on cmnd/%topic%/ but nothing is published to that topic when controlling the blinds from the UI.  I would just like to understand better how the blinds are controlled if no command is sent.

**Comment by barbudor** [comment, marked as answer]
The Web GUI, the Web console, the Serial console are inputs Tasmota core, just like MQTT
So no MQTT publication of the command which somehome remains internal.
But the result is published.
This is true for any command that can be done locally on the GUI such as power on/off or dimmer/pwm control.

**Reply by jenniferlee1818** [reply]
>This is true for any command that can be done locally on the GUI such as power on/off or dimmer/pwm control.

Thank you- I didn't realize that!  

I have the Tasmota integration in Home Assistant and when I use Home Assistant to control my (other) Tasmota devices, the command is published to the MQTT cmnd/%topic%/ but this is not true for my Tasmota blinds, so I'm just wondering why?

From [Home Assistant's Tasmota integration page](https://www.home-assistant.io/integrations/tasmota/):
This integration allows you to control [Tasmota](https://tasmota.github.io/docs/) devices over MQTT.

**Reply by barbudor** [reply]
> nothing is published to that topic when controlling the blinds from the UI.

You may have not been clear here.
My answer applies to Tasmota Web UI.

If you are controlling the blinds from HA, then obvioulsy, HA has no other way to send commands to Tasmota than through MQTT.

**Reply by jenniferlee1818** [reply]
>If you are controlling the blinds from HA, then obvioulsy, HA has no other way to send commands to Tasmota than through MQTT.

Yes, that's what I thought!  My mistake was using MQTT Explorer.  I guess it's not the best client to view the MQTT messages, as it looks like it doesn't update if a topic is published with empty payload.  Switching to MQTT.fx

**Reply by barbudor** [reply]
Both tools have their interrest
MQTT Explorer show a view of the topic tree with latest value for each topic. Also it shows retained topics and allows clearing them.

I don't know MQTT.fx as it is not free but I sometime use [MQTTX](https://mqttx.app/) which is more a logger 
Or just command line tools such as the mosquito-clients (moquitto_sub and mosquitto_pub)
